Mathlib/Order/WellFounded.lean

```diff
@@ -215,45 +215,42 @@ variable (f : α → β)
 
 section LT
 
-variable [LT β] (h : WellFounded ((· < ·) : β → β → Prop))
+variable [LT β] [h : WellFoundedLT β]
 
 /-- Given a function `f : α → β` where `β` carries a well-founded `<`, this is an element of `α`
 whose image under `f` is minimal in the sense of `Function.not_lt_argmin`. -/
 noncomputable def argmin [Nonempty α] : α :=
-  WellFounded.min (InvImage.wf f h) Set.univ Set.univ_nonempty
+  WellFounded.min (InvImage.wf f h.wf) Set.univ Set.univ_nonempty
 
-theorem not_lt_argmin [Nonempty α] (a : α) : ¬f a < f (argmin f h) :=
-  WellFounded.not_lt_min (InvImage.wf f h) _ _ (Set.mem_univ a)
+theorem not_lt_argmin [Nonempty α] (a : α) : ¬f a < f (argmin f) :=
+  WellFounded.not_lt_min (InvImage.wf f h.wf) _ _ (Set.mem_univ a)
 
 /-- Given a function `f : α → β` where `β` carries a well-founded `<`, and a non-empty subset `s`
 of `α`, this is an element of `s` whose image under `f` is minimal in the sense of
 `Function.not_lt_argminOn`. -/
 noncomputable def argminOn (s : Set α) (hs : s.Nonempty) : α :=
-  WellFounded.min (InvImage.wf f h) s hs
+  WellFounded.min (InvImage.wf f h.wf) s hs
 
 @[simp]
-theorem argminOn_mem (s : Set α) (hs : s.Nonempty) : argminOn f h s hs ∈ s :=
+theorem argminOn_mem (s : Set α) (hs : s.Nonempty) : argminOn f s hs ∈ s :=
   WellFounded.min_mem _ _ _
 
--- Porting note (https://github.com/leanprover-community/mathlib4/issues/11119): @[simp] removed as it will never apply
```

Comment:
Removing these porting notes seems orthogonal; I think you should restore them and let someone handle them at all once.